dataset_internal_test.go

```diff
@@ -0,0 +1,64 @@
+package ep
+
+import (
+	"github.com/panoplyio/ep/compare"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestAllMergeOptionResults(t *testing.T) {
+	t.Run("CompareResultEqualMergeOptions", func(t *testing.T) {
+		left := []compare.Result{compare.Equal, compare.Equal, compare.Equal, compare.Equal, compare.Equal}
+		right := []compare.Result{compare.Equal, compare.BothNulls, compare.Null, compare.Greater, compare.Less}
+		expected := []compare.Result{compare.Equal, compare.BothNulls, compare.Null, compare.Greater, compare.Less}
+
+		res := left
+		mergeWith := right
+		merge(res, mergeWith)
+		require.Equal(t, expected, res)
```

Comment:
in case it's 5 separate tests, what do you think about:
```suggestion
		// Equal should always be overridden 
		require.Equal(t, right, res)
```